### DISCUSSION TITLE
Issues with LGCP point abundance estimates, possibly due to SPDE behaviour

### DISCUSSION THREAD

**Discussion by andyhouldcroft** [inlabru-org/inlabru#149] [Q&A] [answered]
Hello all,

I'm working on an LGCP model for distance sampling data, however the population estimates for number of groups in my study site are clearly highly inaccurate (e.g. ~30 groups with mean group size of ~3.6, compared to an expected 7000+ individuals). 

The SPDE appears to be part of the issue and although Finn kindly highlighted this previously, i am not sure how to fix it. I should note that since my last post i have changed "inla.spde.2.pcmatern" to a "inla.spde.2.matern" is this acceptable, i assume this opts to use default priors? (problems with group estimates were occurring before this change). 

I have also included a test model (fit2) with the first of many spatial covariate "SpatialGridDataFrames" to come, is this the correct way to include covariates of this type? Additionally, the model uses the maximum of 10 iterations to converge, is this an issue? 

Sorry for another lengthy post, thank you for any help you can offer. Here is my code and model outputs, I would really appreciate any input on how to generally improve this model.

Andrew 
-----

```
# Set experimental mode 
bru_options_set(inla.mode = "experimental")

# Read in data
campbells <- readRDS("campbells.rds")

# Specify truncation distance 
W <- 60

# Specify half-normal detection function with log transformation explicitly stated 
log_hn <- function(distance, lsig) {
  (-0.5 * (distance / exp(lsig))^2)
}

# Specify SPDE
matern <- inla.spde2.matern(campbells$mesh)

# Specify components
cmp1 <- ~ mySPDE(main = coordinates, model = matern) + lsig(1) + Intercept(1)
cmp2 <- ~ dist_road(campbells$covariates$dist_road, model = "linear") + mySPDE(main = coordinates, model = matern) + lsig(1) + Intercept(1)

# Specify formulas
form1 <- coordinates + distance ~ mySPDE + log_hn(distance, lsig) + Intercept + log(2)
form2 <- coordinates + distance ~ log(1 + dist_road^2) + mySPDE + log_hn(distance, lsig) + Intercept + log(2)

# Fit model without covariate
fit1 <- lgcp(
  components = cmp1,
  campbells$points,
  samplers = campbells$samplers,
  domain = list(
    coordinates = campbells$mesh,
    distance = INLA::inla.mesh.1d(seq(0, 60, length.out = 120))
  ),
  formula = form1,
  options = list(bru_initial = list(lsig = 2), bru_verbose = 4)
)

# Fit model with covariate
fit2 <- lgcp(
  components = cmp2,
  campbells$points,
  samplers = campbells$samplers,
  domain = list(
    coordinates = campbells$mesh,
    distance = INLA::inla.mesh.1d(seq(0, 60, length.out = 120))
  ),
  formula = form2,
  options = list(bru_initial = list(lsig = 2), bru_verbose = 4)
)

# Model output summary
summary(fit1)

_inlabru version: 2.5.2
INLA version: 22.06.03
Components:
  mySPDE: Model types main='spde', group='exchangeable', replicate='iid'
  lsig: Model types main='linear', group='exchangeable', replicate='iid'
  Intercept: Model types main='linear', group='exchangeable', replicate='iid'
Likelihoods:
  Family: 'cp'
    Data class: 'SpatialPointsDataFrame'
    Predictor: 
        coordinates + distance ~ mySPDE + log_hn(distance, lsig) + Intercept + 
            log(2)
Time used:
    Pre = 4.04, Running = 4.19, Post = 0.343, Total = 8.57 
Fixed effects:
            mean    sd 0.025quant 0.5quant 0.975quant   mode   kld
lsig       3.186 0.074      3.041    3.186      3.331  3.186 0.000
Intercept -3.890 4.336    -13.787   -3.966      6.744 -3.989 0.033

Random effects:
  Name	  Model
    mySPDE SPDE2 model

Model hyperparameters:
                   mean    sd 0.025quant 0.5quant 0.975quant  mode
Theta1 for mySPDE  8.04 0.411       7.29     8.01       8.91  7.92
Theta2 for mySPDE -9.32 0.561     -10.53    -9.28      -8.33 -9.12

Deviance Information Criterion (DIC) ...............: -1603.03
Deviance Information Criterion (DIC, saturated) ....: -153568.87
Effective number of parameters .....................: -1891.30

Watanabe-Akaike information criterion (WAIC) ...: 379.48
Effective number of parameters .................: 9.81

Marginal log-Likelihood:  -1124.19 
 is computed 
Posterior summaries for the linear predictor and the fitted values are computed
(Posterior marginals needs also 'control.compute=list(return.marginals.predictor=TRUE)')_

# Abundance estimate
predpts <- ipoints(samplers=campbells$site, domain=campbells$mesh)

Lambda1 <- predict(fit1, predpts, ~ sum(weight * exp(mySPDE + Intercept)))
# Just SPDE to investigate further
Lambda2 <- predict(fit1, predpts, ~ sum(weight * exp(mySPDE))) 

Lambda1

      mean       sd   q0.025   median   q0.975     smin    smax        cv      var
1 28.08559 3.880743 22.19289 27.56347 36.25118 20.66781 44.4857 0.1381756 15.06017

Lambda2

        mean          sd    q0.025   median    q0.975         smin         smax       cv          var
1 9000431645 89280333616 0.8191336 1365.914 207396456 2.509515e-08 892850262968 9.919561 7.970978e+21

?????

```

**Comment by finnlindgren** [comment]
Hi,
just to rule out any potentially already solved bugs since the 2.5.2 CRAN release, can you try it with the development version? (Either from github or r-universe, see https://github.com/inlabru-org/inlabru#readme for installation instructions)

**Comment by finnlindgren** [comment]
`inla.spde2.pcmatern` models tend to be more robust than `inla.spde.matern` models, so you might need to reconsider that change, and more carefully consider the prior parameter settings, especially since the default priors you're using now are unlikely to be a good match to your data. I also note that the intercept parameter seems to not be well constrained by the data.
The non-linear `dist_road` model might also cause issues as it has a bimodal posterior (can't distinguish between positive and negative values of the `dist_road` parameter.

**Reply by finnlindgren** [reply]
With the development version, use `inlabru:::make_track_plots(fit1)[["default"]]` to generate a diagnostics plot for the convergence (or lack thereof). That plot can tell us a bit more about what's going on.

**Reply by andyhouldcroft** [reply]
Hi Finn,

Thank you for the quick response. I have installed the development version and returned to a inla.spde2.pcmatern, however, upon entering ```inlabru:::make_track_plots(fit1)[["default"]]``` the output is ```NULL```

Andrew

**Reply by andyhouldcroft** [reply]
Upon restarting R i have managed to produce the track plots with the following warning messages:
```
Warning messages:
1: Removed 385 row(s) containing missing values (geom_path). 
2: Removed 385 row(s) containing missing values (geom_path). 
3: Removed 385 row(s) containing missing values (geom_path). 
4: Removed 385 row(s) containing missing values (geom_path). 
```

<img width="1431" alt="Screenshot 2022-06-28 at 23 05 17" src="https://user-images.githubusercontent.com/107421097/176308561-69c4e544-84aa-4f92-96fc-5d003fec6545.png">

**Comment by andyhouldcroft** [comment, marked as answer]
The solution to this problem is detailed here:

https://groups.google.com/g/r-inla-discussion-group/c/s96n9bIJZAc/m/OBOEVYl-GAAJ
